src/pool/cli.c

```diff
@@ -1507,6 +1507,12 @@ create_map_refresh_rpc(struct dc_pool *pool, unsigned int map_version,
 		return rc;
 	}
 
+	/* Shorten the timeout to speed up map_refresh tasks. */
+	rc = crt_req_get_timeout(c, &timeout);
+	D_ASSERTF(rc == 0, "crt_req_get_timeout: "DF_RC"\n", DP_RC(rc));
+	rc = crt_req_set_timeout(c, max(1, timeout / 4));
```

Comment:
Just curious, why /4 here?

Reply:
@wangdi1, it is arbitrary. I think we need to work out a set of principles for RPC and event timeouts. (And, that probably should use a short base and multiply for different RPCs/events, instead of dividing.) Here, I'm just making a short-term change to reduce the impact on the time required to complete a map_refresh. 1/8 feels a bit short, 1/2 a bit long...

Reply:
ok, sure. thanks.

Reply:
I am wondering how small is too small for this timeout assignment. For example if one sets the CRT_TIMEOUT of the client to 10 seconds, such as in PR https://github.com/daos-stack/daos/pull/8254,  then I think this will result in a 2 second timeout configured for POOL_TGT_QUERY_MAP RPC. Which I only guess could be problematic in some situations?

I know it doesn't work out so cleanly with my above example of CRT_TIMEOUT=10, but what if the code established a minimum timeout for POOL_TGT_QUERY_MAP of (CRT_DEFAULT_TIMEOUT_US / 4) = 15 seconds based on current code value of CRT_DEFAULT_TIMEOUT_US=60. So the code here might be
crt_req_set_timeout(c, max(CRT_DEFAULT_TIMEOUT_US / 4, timeout / 4));

Does that make sense? I'm looking at this PR as the basis of another one that focuses on MGMT_POOL_FIND RPC and started to consider this idea. https://github.com/daos-stack/daos/pull/7624